### PR TITLE
Feat/initialize sdl2 lib

### LINKED ISSUE #15: Initialize SDL2 Lib
# :white_check_mark: What's needed
Initialize SDL2 in the constructor class

## 📋 Subtasks
- create the constructor and destructor
- create the window

## :file_folder: Files required
src/Graphic/SDL2/

### PR DESCRIPTION
**Pull request by mathieurobert1**
Initialize SDL2 lib:
- constructor to init SDL and window
- destructor to quit SDL
- write empty methods for the rests

/!\ move all Interfaces in an include folder

more details : #15

### CODE REVIEW

**Review by theophile-jr — APPROVED** [review]

File: src/Graphical/SDL2/sdl2.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+#pragma once
+
+#include "AGraphic.hpp"
+#include <SDL2/SDL.h>
+
+class SDL2_lib : public AGraphic {
+
+    private:
+        SDL_Window *_pWindow;
```
Comment (Thomaltarix) [inline src/Graphical/SDL2/sdl2.hpp]: Can it be unique pointers here ?

File: src/Graphical/SDL2/sdl2.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        SDL_Renderer *_pRenderer;
```
Comment (Thomaltarix) [inline src/Graphical/SDL2/sdl2.hpp]: What is the meaning of the 'p' before `Window` or `Renderer` ?

**Review by Thomaltarix — APPROVED** [review]
I commented things, but overall it's great

File: src/Graphical/SDL2/sdl2.hpp (hunk 1 above)
Comment (Thomaltarix) [reply src/Graphical/SDL2/sdl2.hpp]: (Same for the line after)

**Pull request merged**